models/Asset.php

```diff
@@ -1677,20 +1681,23 @@ private function updateChildPaths(FilesystemOperator $storage, string $oldPath,
             $newPath = $this->getRealFullPath();
         }
 
-        try {
-            $children = $storage->listContents($oldPath, true);
-            foreach ($children as $child) {
-                if ($child['type'] === 'file') {
-                    $src  = $child['path'];
-                    $dest = str_replace($oldPath, $newPath, '/' . $src);
+        $children = $storage->listContents($oldPath, true);
+        foreach ($children as $child) {
+            if ($child['type'] === 'file') {
+                $src  = $child['path'];
+                $dest = str_replace($oldPath, $newPath, '/' . $src);
+
+                try {
                     $storage->move($src, $dest);
+                    //TODO: $this->getDao()->updatePath($src); same as DAO->updateChildPaths() but only for that single file that got moved OR collect the single asset in an array and do a mass update, but the former is more anti-dead-lock(?)
+                } catch (UnableToMoveFile $e) {
```

Comment:
maybe just keep track of the failures here and roll them back?

the code as is still has the problem that you might move successfully move some of the files and have others fail but commit to the DB the entire move.

Reply:
Tracking failure would be a good input, at least it should be logged and displayed to the frontend for manual intervention.

The DB would only commit the single moved files (by queries that updates 1 entry, see the TODO updatePaths())  and keep the unmoved children in the former path physically and in DB. 

So at the end, source and destination folders would co-exists but it is still considered a successful rename with just some "leftover". It would be intended that on failure when at least one file is moved, it doesn't rollback, so treating the failed cases as a literal exception in the process.

Reply:
Mhh the query could also throw an exception and then the file should be moved back. In the end, they both need to match the actual path.

Reply:
Imho the entire "update" transaction needs to be rolled back in case something goes wrong. Otherwise we wouldn't have data-loss, yes, but still it's not consistent. 

Created #17850 as an alternative, WDYT?